connectors/sources/generic_database.py

```diff
@@ -0,0 +1,444 @@
+                                f"{query_kwargs['table']}_{column}".lower()
+                                for column in cursor.keys()
+                            ]
+                        yield_once = False
+
+                    while True:
+                        rows = cursor.fetchmany(size=size)
+                        rows_length = len(rows)
+
+                        if not rows_length:
+                            break
+
+                        for row in rows:
+                            yield row
+
+                        rows_fetched += rows_length
+                        await asyncio.sleep(0)
+                else:
+                    yield cursor.fetchall()
+                break
+            except InternalClientError:
+                raise
+            except Exception as exception:
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                if retry == self.retry_count:
+                    raise exception
+                await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry)
+                retry += 1
+
+    async def _async_connect(self, query, engine):
+        """Execute the passed query on the Async supported Database server and return cursor.
+
+        Args:
+            query (str): Database query to be executed.
+            engine (str): Database engine to be used.
+
+        Returns:
+            cursor: Asynchronous cursor
+        """
+        try:
+            async with engine.connect() as connection:
+                cursor = await connection.execute(text(query))
+                return cursor
+        except Exception as exception:
+            logger.warning(
+                f"Something went wrong while executing query. Exception: {exception}"
+            )
+            raise
+
+    async def close(self):
+        """Close the connection to the database server."""
+        if self.connection is None:
+            return
+        self.connection.close()
+
+    async def _sync_connect(self, query, engine):
+        """Executes the passed query on the Non-Async supported Database server and return cursor.
+
+        Args:
+            query (str): Database query to be executed.
+            engine (Engine): Engine to connect with the Database server
+
+        Returns:
+            cursor: Synchronous cursor
+        """
+        try:
+            loop = asyncio.get_running_loop()
+            self.connection = await loop.run_in_executor(
+                executor=None, func=engine.connect
+            )
+            cursor = await loop.run_in_executor(
+                executor=None, func=partial(self.connection.execute, statement=query)
+            )
+            return cursor
+        except Exception as exception:
+            logger.warning(
+                f"Something went wrong while executing query. Exception: {exception}"
+            )
+            raise
+
+    def serialize(self, doc):
```

Comment:
this is being copy-pasted in many sources at this point.

Let's move it to `BaseDataSource` to avoid duplication (and remove it from mongo, MySQL)